src/manifest.c

```diff
@@ -323,6 +322,20 @@ static struct manifest *manifest_from_file(int version, char *component, bool he
 		c = c2;
 
 		file->filename = strdup_or_die(c);
+		file->bundlename = strdup_or_die(c);
+		/* If the file is an iterative manifest, copy the from_version and rename the
+		 * manifest so it matches the bundle name */
+		if (file->is_iterative == 1) {
+			char *ptr;
+			ptr = strrchr(file->filename, '.');
+			if (ptr != NULL) {
```

Comment:
If ptr is NULL we aren't causing any error, but from_version won't be set. I think we should raise an error for this Manifests, because we aren't going to be able to use it.

Reply:
good point.

Reply:
Ok, but what if:
* manifest is given an `I` flag
* manifest is named `name.F.onebillion`

Looks like this manifest well be loaded as an iterative manifest with a `from` version of `0`. In this case we should certainly gracefully fall back to the full manifest. Although in the above example we will fallback to the full manifest (right?) because of the `0` `from` version, but what about if the manifest is named `name.F.22940`? Do we want to continue to try to use the improperly named manifest?

Reply:
If you think this is something that can actually happen in production (to have a bundle with a name like that) then we definitely need to validate the name, so we make sure we are loading the correct from_version.
As you mentioned, if we find that the name is incorrect or we couldn't determine the from_version, we can always just add a from_version of 0 which would cause the iterative manifest to be discarded later.

Reply:
Let's forward-proof this code so in case we add a different type of manifest in the future (like `name.F.22590`) we don't need a format bump to introduce the feature.

Reply:
ok, let's see what I can do with this